From ee1cfc844e85489a83c052fd3fd67b9e79c14148 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 25 Nov 2014 12:08:48 +0800
Subject: [PATCH 0147/5242] MLK-9897-2 usb: otg-fsm: check B-device state
 before sending polling request

commit  a4dd6e88d38cbaef787f6be01109e15b0a78983f from
https://source.codeaurora.org/external/imx/linux-imx.git

While system resume, the connected udev as B-device may not has been resumed,
if HNP polling request is sent to it, HNP polling may fail, this patch adds
check of the udev state to make sure it finished resume and to be configured
state before sending host request message, otherwise try next HNP polling
request.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <jun.li@freescale.com>
(cherry picked from commit 2cff5eb7bd4feb70cd28ac4655e7433e57a17938)
(cherry picked from commit b6eea060a3760bdb24d1514a5130a346daba81c9)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/common/usb-otg-fsm.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 88c29a9..f0538f5 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -144,6 +144,13 @@ static void otg_hnp_polling_work(struct work_struct *work)
 		return;
 	}
 
+	if (udev->state != USB_STATE_CONFIGURED) {
+		dev_dbg(&udev->dev, "the B dev is not resumed!\n");
+		schedule_delayed_work(&fsm->hnp_polling_work,
+				      msecs_to_jiffies(T_HOST_REQ_POLL));
+		return;
+	}
+
 	/*
 	 * Legacy otg test device does not support HNP polling,
 	 * start HNP directly for legacy otg test device.
-- 
1.7.9.5

